website/blog/unusual-builds-w-bytes.md

```diff
@@ -0,0 +1,224 @@
+The PR explored a potential workaround option for this issue. However, it was incomplete and was never reviewed.
+
+## The Workaround
+
+As Bazel 8.2.0 and 7.6.0 have not been released yet, users on older versions of Bazel can work around this issue by setting the `--experimental_remote_cache_ttl` flag to a very long value (i.e., `10000d` for 27 years) to effectively disable the TTL feature.
+
+> I hope you don't keep your Bazel running for that long! 😅
+
+However, there are some caveats to this workaround:
+It's possible for the local blob metadata to become out-of-sync with the Remote Cache.
+In particular, at BuildBuddy, we employ an LRU eviction scheme to keep the Remote Cache size under control.
+With a longer TTL set by Bazel, Bazel will also trust that our Remote Cache will keep the cache data around for that long, which we do not guarantee.
+So, it's possible that the Remote Cache will evict the blob data before the TTL expires, which can cause Bazel to make incorrect assumptions and result in errors such as:
+
+```bash
+remote cache evicted: ...
+    com.google.devtools.build.lib.remote.common.BulkTransferException: 3 errors during bulk transfer:
+        com.google.devtools.build.lib.remote.common.CacheNotFoundException: Missing digest: d0387e622e30ab61e39b1b91e54ea50f9915789dde7b950fafb0863db4a32ef8/17096
+        com.google.devtools.build.lib.remote.common.CacheNotFoundException: Missing digest: 9718647251c8d479142d459416079ff5cd9f45031a47aa346d8a6e719e374ffa/28630
+        com.google.devtools.build.lib.remote.common.CacheNotFoundException: Missing digest: 785e0ead607a37bd9a12179051e6efe53d7fb3eb05cc291e49ad6965ee2b613d/11504
+```
+
+I have documented this problem in detail in our [Troubleshooting RBE Failures docs](https://www.buildbuddy.io/docs/troubleshooting-rbe#cachenotfoundexception-missing-digest).
+The TLDR is that you can use a couple of flags together with the TTL flag to mitigate this issue:
+
+```bash
+# Set a long TTL to avoid excessive downloads.
+# Not needed after Bazel 8.2.0.
+# Reference: https://github.com/bazelbuild/bazel/pull/25398
+common --experimental_remote_cache_ttl=10000d
+
+# If the Remote Cache evicts the blob data before the TTL expires,
+# Bazel will reset itself, dropping all the in-memory metadata before
+# retrying the entire build. The value controls how many times Bazel
+# would retry before giving up.
+#
+# Default changed from 0 to 5 since Bazel 8.0.0.
+common --experimental_remote_cache_eviction_retries=5
+```
+
+We are also improving the reliability of eviction retry via [PR#25358](https://github.com/bazelbuild/bazel/pull/25358) and [PR#25448](https://github.com/bazelbuild/bazel/pull/25448) for future Bazel releases.
+As a longer term strategy, we are researching proper way to implement Action Rewinding in Bazel so that missing artifacts can be recreated seamlessly by re-running the action that produced them.
```

Comment:
```suggestion
As a longer term strategy, we are researching ways to add support for Action Rewinding to Bazel so that missing artifacts can be recreated seamlessly within a single build by re-running the actions that produced them.
```

Reply:
Done